src/commands/dev/index.ts

```diff
@@ -409,6 +412,52 @@ export default class Dev extends BaseCommand {
     return parsed;
   }
 
+  /**
+   * Cleanup architect-created images and layers that are no longer in use to prevent
+   * old images from wasting space on disk.
+   */
+  async pruneImages(compose: DockerComposeTemplate): Promise<void> {
+    // Remove build cache layers with our label that are older than 7 days
+    await docker(['builder', 'prune', '--filter', `label=${DOCKER_IMAGE_LABEL}`, '--filter', `unused-for=${7 * 24}h`, '-af'], { stdout: false });
+    // Remove images that are untagged and created by previous `architect dev` runs
+    await docker(['image', 'prune', '--filter', `label=${DOCKER_IMAGE_LABEL}`, `--filter`, 'dangling=true', '-f'], { stdout: false });
+
+    // Find the set of services and name of all components being deployed. Any image that is labelled with the name
+    // of a component being run but uses a Repository name that no longer matches any service names can be
+    // removed. This can happen when changing the name of a service in the architect.yml.
+    const services = new Set();
+    const components = [];
+    for (const [service_name, service] of Object.entries(compose.services)) {
+      services.add(service_name);
+      for (const label of service.build?.labels || []) {
+        if (label.startsWith('component=')) {
+          components.push(label.split('=')[1]);
+        }
+      }
+    }
+
+    for (const component_name of components) {
+      const built_image_result = await docker(['image', 'list',
+        '--filter', `label=${DOCKER_IMAGE_LABEL}`,
+        '--filter', `label=component=${component_name}`,
```

Comment:
Is there a way to do something like `'--filter', `label=component=component_name_1|component=component_name_2|component=component_name_3` to reduce the amount of Docker CLI calls we're waiting on?

Reply:
good q I'll look into this, not sure if filters behave as OR/AND and that can maybe prevent the docker call in a loop 👍

Reply:
Did some poking around and it seems like these filters are always ANDed together so there's not a way to combine commands. That being said, this situation only comes up if you `architect dev component_1.yml component_2.yml ...` which is fairly uncommon, most cases involve running a single component so this loop has an N = 1